lib/ansible_tower_client/collection.rb

```diff
@@ -11,24 +11,34 @@ def all
     end
 
     def collection_for(paginated_result)
-      body = JSON.parse(paginated_result.body)
+      body = YAML.safe_load(paginated_result.body)
       results = body["results"]
       loop do
         break if body["next"].nil?
-        body = JSON.parse(api.get(body["next"]).body)
+        body = YAML.safe_load(api.get(body["next"]).body)
         results += body["results"]
       end
 
-      build_collection(results)
+      build_collection(hashify(results))
```

Comment:
@bdunne - Are you thinking something more like `hashify(results, [:extra_vars, :survey_spec])` or
`hashify(results, EXCLUDED)`